### PR TITLE
[Feature] Add YOLOXLrUpdaterHOOK

### PR DESCRIPTION
**Pull request by HsLOL**
## Motivation

Add YOLOXLrUpdaterHOOK class which is used in YOLOX.

### CODE REVIEW

File: mmdet/core/hook/cosinesnnealingwithstop_lrupdater_hook.py [hunk 1]
```diff
@@ -0,0 +1,62 @@
+from mmcv.runner.hooks.lr_updater import CosineAnnealingLrUpdaterHook, annealing_cos
+from mmcv.runner.hooks import HOOKS
+from mmcv.runner import get_dist_info
+
+
+@HOOKS.register_module()
+class YOLOXLrUpdaterHOOK(CosineAnnealingLrUpdaterHook):
+    """There are two main differences between YOLOXLrUpdaterHOOK
```
Comment (ZwwWayne) [inline mmdet/core/hook/cosinesnnealingwithstop_lrupdater_hook.py:8]: Learning rate updater for YOLOX.

There are two main.....

File: mmdet/core/hook/yolox_lrupdater_hook.py [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+
+
+@HOOKS.register_module()
+class YOLOXLrUpdaterHook(CosineAnnealingLrUpdaterHook):
+    """YOLOX learning rate scheme.
+
+    There are two main differences between YOLOXLrUpdaterHook
+    and CosineAnnealingLrUpdaterHook.
+
+    1. When the current running epoch is greater than
```
Comment (jshilong) [inline mmdet/core/hook/yolox_lrupdater_hook.py:14]: add  indentation here

File: mmdet/core/hook/yolox_lrupdater_hook.py [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+    Args:
+        last_epoch (int): The number of last epoch with a fixed learning rate.
+        warmup_ratio (float): LR used at the beginning of warmup.
+           This parameter does not depend on the number of GPUs, so we need
+           to multiply by work_size.
+    """
+
+    def __init__(self, last_epoch, warmup_ratio, **kwargs):
+        _, work_size = get_dist_info()
+        self.base_lr_ = warmup_ratio * work_size
```
Comment (jshilong) [inline mmdet/core/hook/yolox_lrupdater_hook.py:27]: Is the final underscore necessary?

Comment (hhaAndroid) [reply mmdet/core/hook/yolox_lrupdater_hook.py:27]: Because self.base_lr is already occupied internally.

Comment (ZwwWayne) [reply mmdet/core/hook/yolox_lrupdater_hook.py:27]: then leave a comment here to indicate the motivation

**Comment by ZwwWayne** [comment]
Need to resolve conflicts before merging this PR.

File: mmdet/core/hook/yolox_lrupdater_hook.py [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+
+    There are two main differences between YOLOXLrUpdaterHook
+    and CosineAnnealingLrUpdaterHook.
+
+       1. When the current running epoch is greater than
+           `max_epoch-last_epoch`, a fixed learning rate will be used
+       2. The exp warmup scheme are different with LrUpdaterHook in MMCV
+
+    Args:
+        last_epoch (int): The number of last epoch with a fixed learning rate.
```
Comment (jshilong) [inline mmdet/core/hook/yolox_lrupdater_hook.py:19]: The number of epochs with a fixed learning rate before the end of the training

**Review by hhaAndroid — APPROVED** [review]

**Review by ZwwWayne — APPROVED** [review]

**Comment by OpenMMLab-Coodinator** [comment]
Hi @HsLOL ！First of all, we want to express our gratitude for your significant PR in this project. Your contribution is highly appreciated, and we are grateful for your efforts in helping improve this open-source project during your personal time. We believe that many developers will benefit from your PR.

We would also like to invite you to join our Special Interest Group (SIG) private channel on Discord, where you can share your experiences, ideas, and build connections with like-minded peers. To join the SIG channel, simply message moderator— OpenMMLab on Discord or briefly share your open-source contributions in the #introductions channel and we will assist you. Look forward to seeing you there! Join us ：https://discord.gg/UjgXkPWNqA

If you have WeChat account，welcome to join our community on WeChat. You can add our assistant ：openmmlabwx. Please add "mmsig + Github ID" as a remark when adding friends：）
Thank you again for your contribution❤
